scripts/import/probabilities/marsh_probabilities.py

```diff
@@ -0,0 +1,443 @@
+        MySQLdb.Error
+            If any error occurs during the database connection or query execution.
+        
+        Example:
+        ----------
+        insert_details_into_meta('localhost', 3306, 'mydatabase', 'password123', 'root')
+
+        This inserts a record into the 'meta' table of the 'mydatabase' database using the current timestamp, 
+        a static description, and source details retrieved from the database.
+
+    """
+    
+    source_id = get_source_details(host, port, db, password, user)
+    description = "Baydoni & Marsh probabilities"
+    
+    insert_into_meta_query = """ INSERT into meta(`key`, date_update, description, version, source_id, is_public) VALUES (%s, %s, %s, %s, %s, %s)
+
+"""
+    #current_datetime = datetime.now()
+    meta_key = KEY + "_" + attrib
+
+    #formatted_datetime = current_datetime.strftime('%Y-%m-%d %H:%M:%S.%f')
+    version = 1
+    database = MySQLdb.connect(host=host,port=port,user=user,passwd=password,db=db)
+    cursor = database.cursor()
+    
+    cursor.execute(insert_into_meta_query, (meta_key, datetime.datetime.now(), description, version, source_id, 0))
+    cursor.close()
+    database.close()
+
+    
+
+def get_source_details(host, port, db, password, user):
+    """
+        Retrieves the source ID from the 'source' table in the database.
+
+        This function connects to a MySQL database and retrieves the `id` of the source where the `name` 
+        matches 'Marsh Mechanism probabilities'. It returns the corresponding `source_id`.
+
+        Parameters:
+        ----------
+        host : str
+            The hostname or IP address of the MySQL database server.
+        
+        port : int
+            The port number used to connect to the MySQL database server.
+        
+        db : str
+            The name of the MySQL database containing the 'source' table.
+        
+        password : str
+            The password to authenticate the MySQL user.
+        
+        user : str
+            The username to authenticate with the MySQL database server.
+
+        Returns:
+        ----------
+        source_id : int
+            The ID of the source with the name 'Marsh Mechanism probabilities'.
+
+        Raises:
+        ----------
+        MySQLdb.Error
+            If any error occurs during the database connection or query execution.
+
+        Example:
+        ----------
+        source_id = get_source_details('localhost', 3306, 'mydatabase', 'password123', 'root')
+
+        This retrieves the `id` from the 'source' table where `name` equals 'Marsh Mechanism probabilities'
+        and returns the corresponding `source_id`.
+    """
+
+    get_source_query = """ SELECT id from source where name = 'Marsh Mechanism probabilities'"""
+
+    database = MySQLdb.connect(host=host,port=port,user=user,passwd=password,db=db)
+    cursor = database.cursor()
+
+    cursor.execute(get_source_query)
+    source_id = cursor.fetchone()
+    source_id = source_id[0]
```

Comment:
Before fetching the results, this method should fail if the source cannot be found.